newsLetter/newsletter.py

```diff
@@ -255,6 +257,101 @@ def format_newsletter(content: str)->tuple[str, str]:
     newsletter_content += "Stay informed with Wall-E's tech updates, and see you back here tomorrow!\n\n---\n\n[🔊 Listen to the Full Podcast Episode Here](https://aibriefingroom.podbean.com/)"
     return newsletter_content
 
+def email_exists(service, email: str) -> bool:
+    SPREADSHEET_ID = os.getenv(SPREADSHEET_SRC)
+    RANGE_NAME = 'response!B:B'  # Assuming email is in column B
+
+    result = service.spreadsheets().values().get(
+        spreadsheetId=SPREADSHEET_ID,
+        range=RANGE_NAME
+    ).execute()
+
+    emails = result.get('values', [])
+    emails_flat = [email_row[0] for email_row in emails if email_row]  # Flatten the list
+    return email in emails_flat
+
+def signup_newsletter(service, name: str, email: str, preferences: list, ) -> None:
+    SPREADSHEET_ID = os.getenv(SPREADSHEET_SRC)
+    RANGE_NAME = 'response!A:F'
+
+    # Unsubscribed UUID
+    def generate_unsubscribed_uuid():
+        return str(uuid.uuid4())
+
+    # Prepare the values to be appended
+    date = datetime.now().strftime('%m/%d/%Y')
+    values = [[date, email, name, ', '.join(preferences), "", generate_unsubscribed_uuid()]]
+    body = {'values': values}
+    result = service.spreadsheets().values().append(
+        spreadsheetId = SPREADSHEET_ID, 
+        range=RANGE_NAME, 
+        valueInputOption='RAW', 
+        insertDataOption="INSERT_ROWS", 
+        body=body
+    ).execute()
+
+    print(f"Added {name} to the newsletter list...")
+
+
+""" Unsubscribe by email (deprecated) """
+def unsubscribe_user(service, email: str) -> bool:
```

Comment:
Remove this function since it is not used? Unless there is a reason behind keeping it.

Reply:
Let's keep it until we start building the feature?